draft.md

```diff
@@ -0,0 +1,81 @@
+# Draft Ipld Unixfs Spec
+
+## Basic Structure
+
+   - Some sort of header that indicates that this a directory and included a version number.  The header could also have fields to give additional information on the meaning of the extended attributes.
+
+  - CBOR Map
+    - Key: CBOR Byte or Text String: File Name
+    - Value: CBOR Array of:
+      - Type: CBOR Unsigned Int
+      - Link or Data: CBOR Type varies
+      - Optional file size: CBOR Unsigned Int
+      - Optional Standard Attributes: CBOR Map
+      - Optional Extended Attributes: CBOR Map
+
+The file size is only defined for regular files and is the size of the file contents.
+
+All maps should be ordered based on the binary values of the key,
+duplicates are not allowed.
+
+### Notes
+
+ * An array makes sense to be as this is more compact and the value of
+   the fields are unambiguous, it also allows for a separation of
+   standard and extended attributes
+
+ * The key type can either be a byte or text string as POSIX makes no
+   requirements that file names be utf-8 and it is important that any
+   file name can be faithfully represented, if the string is utf-8
+   then the type will be Text.
+
+## Types
+
+The type field should be limited to a set of well defined values so it
+makes sense that this is an integer rather than a text string.  The
+value is the ascii value of a letter. When converting to JSON the
+integer can be represented as a single character string.
+
+Possible values are as follows:
+
+  * 0, '', `file`: regular file
+  * `e`, `exe`: executable file
+  * `d`, `dir`: directory entry
+  * `s`, `special`: special file type (fifo, device, etc). The second field is a CBOR Map with at least one field to describe the type.
+  * `l`, `symlink`: symbolic link.  The second field is the contents of the link
+  * `o`, `other`: link to other ipld object, links followed for GC and related operations 
+  * `u`, `unknown`: link to unknown objects, links not followed
+
```

Comment:
If an integer enumeration was used rather than ascii characters, the canonical CBOR representation would be packed to one byte rather than two. Given that the CID will be in raw representation, I don't think clarity would suffer by an enumeration.

Reply:
I am not agents this.